### PR TITLE
Enhance Player Profile Management

### PR DESCRIPTION
**Pull request by Xskullibur**
- Updates player profile management, including profile picture support and DTO enhancements.
- Refactors authentication services and JWT filter logic.
- Adds custom exceptions for file handling.
- Updates security configurations to enforce authentication on relevant endpoints.

### CODE REVIEW

File: frontend/app/profile/edit/page.tsx [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+          });
+          router.push("/profile/" + user?.userId);
+        }
+      })
+      .catch((error: AxiosError) => {
+        let title = "Error";
+        let message = error.message;
+
+        if (error.response) {
+          switch (error.response.status) {
```
Comment (ceewaigit) [inline frontend/app/profile/edit/page.tsx:103]: should refactor and separate this message into alert dialog

File: frontend/app/profile/edit/page.tsx [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+      profileUpdates: {
+        ...prev.profileUpdates,
+        profilePicturePath:
+          PROFILE_IMAGE_API + playerProfileRequest.profileUpdates.username,
+      },
+      profileImage: new File([], ""),
+    }));
+  };
+
+  if (isLoading) return <div>Loading...</div>;
```
Comment (ceewaigit) [inline frontend/app/profile/edit/page.tsx:182]: i might do a loading animation screen and we can just insert the animation screen

File: frontend/app/profile/edit/page.tsx [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+              <label htmlFor="firstName" className="block text-xl mb-2">
+                First Name:
+              </label>
+              <input
+                type="text"
+                id="firstName"
+                name="firstName"
+                value={playerProfileRequest.profileUpdates.firstName || ""}
+                onChange={handleChange}
+                className="w-full p-2 rounded-lg bg-[#333333] text-white"
```
Comment (ceewaigit) [inline frontend/app/profile/edit/page.tsx:214]: shouldnt hardcode bg-[hexcolor], should change to use bg-secondary or bg-primary-foreground or something

**Review by ceewaigit — APPROVED** [review]
apart from my nitpicks, lgtm

File: frontend/app/profile/edit/page.tsx (hunk 1 above)
Comment (Xskullibur) [reply frontend/app/profile/edit/page.tsx:103]: Will be refactored in the next sprint

File: frontend/app/profile/edit/page.tsx (hunk 3 above)
Comment (Xskullibur) [reply frontend/app/profile/edit/page.tsx:214]: will be refactored in the next sprint

**Pull request merged**